src/Jenssegers/Mongodb/Model.php

```diff
@@ -202,11 +201,11 @@ protected function getDateFormat()
     /**
      * Get a fresh timestamp for the model.
      *
-     * @return MongoDate
+     * @return MongoDB\BSON\UTCDateTime
      */
     public function freshTimestamp()
-    {
-        return new MongoDate;
+    {	
+	    return round(microtime(true) * 1000);
```

Comment:
In this case, it gives an exception because that the object values are not same between MongoDate and UTCDateTime, but the script will change generated microtime to an UTCDateTime instance later.